pkg/build/rpm.go

```diff
@@ -0,0 +1,55 @@
+package build
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+)
+
+func (b *Builder) getRPMFileNames() error {
+	b.rpmSourceDir = filepath.Join(b.buildConfig.ImageConfigDir, "rpms")
+
+	rpms, err := os.ReadDir(b.rpmSourceDir)
+	if err != nil {
+		return fmt.Errorf("reading rpm source dir: %w", err)
+	}
+
+	for _, rpmFile := range rpms {
+		b.rpmFileNames = append(b.rpmFileNames, rpmFile.Name())
+	}
+
+	return nil
+}
+
+func (b *Builder) copyRPMs() error {
+	err := b.getRPMFileNames()
+	if err != nil {
+		return fmt.Errorf("getting rpm file names: %w", err)
+	}
+
+	for _, rpm := range b.rpmFileNames {
+		sourcePath := filepath.Join(b.rpmSourceDir, rpm)
+		destPath := filepath.Join(b.combustionDir, rpm)
+
+		sourceFile, err := os.Open(sourcePath)
+		if err != nil {
+			return fmt.Errorf("opening rpm source path: %w", err)
+		}
+
+		destFile, err := os.Create(destPath)
+		if err != nil {
+			return fmt.Errorf("opening rpm dest path: %w", err)
+		}
+
+		_, err = io.Copy(destFile, sourceFile)
+		if err != nil {
+			return fmt.Errorf("copying rpm: %w", err)
+		}
+
+		sourceFile.Close()
```

Comment:
While this will not trigger the linter, it is not the approach which should be taken.

What will happen if `io.Copy()` fails in this function? Both `sourceFIle` and `destFile` will remain open and leak memory.

The proper thing to do is to encapsulate each iteration in a separate function which in turn would be able to leverage `defer` statements and properly free up the resources.

You could either to that or take a look at `filepath.Walk` which will require you to execute a separate function for each call.

Reply:
oh I see, I forgot to consider that, thank you for pointing that out